.github/workflows/main.yml

```diff
@@ -28,6 +28,10 @@ jobs:
     timeout-minutes: 60
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 300
+      - name: Fetch tags
+        run: git fetch --depth=300 origin +refs/tags/*:refs/tags/*
```

Comment:
I've left the default parameters of `actions/checkout@v2` (which is `--depth=1 --no-tags`) on the test runners for performance reasons, so that it doesn't fetch unnecessary files from the git history, like the ffmpeg binary file (~30MiB) for example that we've now finally removed. Fetching all tags also means fetching this file.

I am aware that versioneer (or `git describe --tags`) doesn't work if there is not a single tag in the cloned repo, but is there a reason why the test runners need a correct version string? All that's important is the commit ID.

The depth value of `300` on all the additional "Fetch tags" commands should be set back to `1`, as it's either unnecessary or redundant. It is there to limit fetching more commits than the fetch depth value defines in the `actions/checkout@v2` action, but it'll still fetch all tags, regardless the depth value. Unfortunately, you can't fetch just the latest tag with the current git version, which would be the perferred thing to do here. If we **really** need only the latest tag, then it could be read from the Github API and the additional command be put into a local action inside the `.github` directory for DRY reasons.

Reply:
@back-to What's your opinion here? Do you want to keep this or can this part be reverted?

I'd really like to merge the other changes, but if you think that we need to fetch the tags on all test runners, then I don't want to block this. I'm just saying that it's unnecessary, as I don't see any negative impact when fetching only one commit other than having a couple of "untagged" lines in the parts of the build log that are not relevant to the test runs and coverage reports.